src/erlazure_insights.erl

```diff
@@ -0,0 +1,130 @@
+-module(erlazure_insights).
+
+-export([
+        get_events/3,
+        get_time_offset_as_timestamp/2,
+        get_events_all/3,
+        follow_token/3,
+        create_time_bounded_filter/2
+    ]).
+
+-include("erlazure.hrl").
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-export([test_url_construction/0]).
+-endif.
+
+-type datetime() :: {{number(), number(), number},
+    {number(), number(), number()}}.
+
+-type offset() :: {seconds | minutes | hours, number()}.
+
+-spec get_events_all(string(), string(), #azure_config{}) ->
+    {ok, list()} | {{error, term()}, {accumulated, list()}}.
+get_events_all(SubscriptionId, Filter, Config=#azure_config{}) ->
+    {ok, Values, NextToken} = get_events(SubscriptionId, Filter, Config),
+    case NextToken of
+        undefined -> {ok, Values};
+        _ -> follow_token(binary_to_list(NextToken), Values, Config)
+    end.
+
+-spec get_events(string(), string(), #azure_config{}) ->
+    {ok, list(), undefined | binary()}.
+get_events(SubscriptionId, Filter, Config=#azure_config{}) ->
```

Comment:
`get_events` should be able to accept `SkipToken` as well - since this is the one of the parameters which endpoint https://msdn.microsoft.com/en-us/library/azure/dn931934.aspx accepts.